lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -418,19 +445,27 @@ class QcFlagService {
             const removedFlag = await QcFlagRepository.removeOne({ where: { id } });
 
             // Reconstruct flags ineffectiveness periods
-            const commonDataOrSimulationPassInclude = [ // QC flag can be associated with only one data pass or only one simulation pass
-                qcFlag.dataPasses?.length > 0
-                    ? { association: 'dataPasses', where: { id: qcFlag.dataPasses[0].id } }
-                    : { association: 'simulationPasses', where: { id: qcFlag.simulationPasses[0].id } },
-            ];
+            const flagIncludes = [];
+            let synchronousQcWhereClause = null;
+            // QC flag can be associated with only one data pass or only one simulation pass or be synchronous
+            if (qcFlag.dataPasses?.[0]?.id) {
+                flagIncludes.push({ association: 'dataPasses', where: { id: qcFlag.dataPasses[0].id }, required: true });
+            } else if (qcFlag.simulationPasses?.[0]?.id) {
+                flagIncludes.push({ association: 'simulationPasses', where: { id: qcFlag.simulationPasses[0].id }, required: true });
+            } else {
+                flagIncludes.push({ association: 'dataPasses', required: false });
+                flagIncludes.push({ association: 'simulationPasses', required: false });
+                synchronousQcWhereClause = { '$flag->dataPasses.id$': null, '$flag->simulationPasses.id$': null };
+            }
 
             const flagsCreatedAfterRemovedFlag = await QcFlagRepository.findAll({
                 where: {
                     dplDetectorId: removedFlag.dplDetectorId,
                     runNumber: removedFlag.runNumber,
                     createdAt: { [Op.gt]: qcFlag.createdAt },
+                    ...synchronousQcWhereClause ?? {},
```

Comment:
If you coalesce it to {} you should directly set `let synchronousQcWhereClause = {}`

Reply:
applied